client/src/main/java/io/split/storages/memory/InMemoryCacheImp.java

```diff
@@ -206,8 +206,10 @@ private void removeFromFlagSets(String featureFlagName, HashSet<String> sets) {
         if (sets != null) {
             for (String set : sets) {
                 HashSet<String> features = _flagSets.get(set);
-                features.remove(featureFlagName);
-                _flagSets.put(set, features);
+                if (features != null) {
+                    features.remove(featureFlagName);
+                    _flagSets.put(set, features);
```

Comment:
Check Nadia, but maybe you can remove this line (`_flagSets.put(set, features);`), because `features` is a reference already inside the `_flagSets` ConcurrentMap.

Reply:
yes, you are right. I removed the extra line.